### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by iamsjunaid**
Pull Request Summary:

This pull request enhances the User and Post models by implementing essential validations and unit tests to ensure data integrity and accuracy. The changes made in this pull request are as follows:

**User Model:**
- Added validation to ensure that the "Name" field is not left blank.
- Implemented a validation for the "PostsCounter" field to enforce it being an integer greater than or equal to zero.
- Included corresponding unit tests to verify the correctness of the above validations.

**Post Model:**
- Incorporated validation to prevent empty "Title" fields.
- Enforced a maximum length of 250 characters for the "Title" field.
- Introduced validations for "CommentsCounter" and "LikesCounter" to ensure they are integers greater than or equal to zero.
- Added comprehensive unit tests for all the model methods and validations to validate their behavior and accuracy.

This pull request addresses the need for robust data validation and integrity by implementing stringent checks on crucial fields in both the User and Post models. Additionally, the inclusion of thorough unit tests ensures that all model methods and validations perform as intended and safeguard against future regressions.

These enhancements significantly contribute to maintaining a reliable and well-structured data model while promoting code quality and stability.

### CODE REVIEW

File: spec/model/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      end
+    end
+
+    context 'when posts_counter is positive' do
+      it 'returns true' do
+        expect(subject).to be_valid
+      end
+    end
+  end
+end
```
Comment (TedLivist) [inline spec/model/user_spec.rb]: - [x] All model methods should be tested. Kindly add tests for the `recent_posts` method in your User model.

File: spec/model/model_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    it 'is invalid with a non-integer likes_counter' do
+      author = User.create(name: 'Tom & Jerry', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Best friends')
+      post = Post.new(title: 'Example title', text: 'Example body', likes_counter: 'not_an_integer', author:)
+
+      expect(post).to_not be_valid
+      expect(post.errors[:likes_counter]).to include('is not a number')
+    end
+  end
+end
```
Comment (TedLivist) [inline spec/model/model_spec.rb]: - [x] Kindly add tests for the `recent_comments` and `update_posts_counter` methods in the Post model.

File: spec/model/model_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
```
Comment (TedLivist) [inline spec/model/model_spec.rb]: - [x] Kindly add tests for the Like and Comment models. All models should be tested. Remember to add tests for the methods within those models.

- [x] **[OPTIONAL]** Kindly rename this test file to something like `post_spec.rb` as that is a more descriptive name for the tests it contains.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @iamsjunaid,

Good job so far!
✔️ Implementation of validations and specs for them
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by iamsjunaid** [comment]
Hi @TedLivist 😊, Thanks for the thorough review of this PR. I have made all the requested changes⭐.

**Review by Munsa1 — APPROVED** [review]
Hi @iamsjunaid, 
congrats 4

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Passing Tests :heavy_check_mark: 
All Models Tested :heavy_check_mark: 
 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by iamsjunaid** [comment]
Thanks for approving the PR @Munsa1 🙂!!!

**Pull request merged**